File: leetcode_learning/code_py/9.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        # return self.judge_string(x)
        return self.judge_int(x)
    
    # 判断字符串首尾是否相同
    def judge_string(self, x):
        if x<0:
            return False
        str_x = str(x)
        i = 0
        j = len(str_x) - 1
        while(i<j):
            if (str_x[i] != str_x[j]):
                return False
            i += 1
            j -=1
        return True

    # 判断整数反转后是否和原数相同
    def judge_int(self, x):
        if x<0:
            return False
        flip_x = 0
        mod = 0
        store_x = x
        while(x!=0):
            mod = x%10
            flip_x = flip_x*10 + mod
            x //= 10
        return flip_x==store_x

if __name__=='__main__':
    s = Solution()
    x = 1001
    print(s.isPalindrome(x))

File: leetcode_learning/code_cpp/9.cpp
//解法一、判断反转后的数和原数是否相同
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0)
            return 0;
        long flip = 0;
        long temp = 0;
        int store_x = x;
        while(x!=0)
        {
            temp = x%10;
            flip = 10*flip + temp;
            x = x/10;
        }
        return flip==store_x;
    }
};

//解法二、使用字符串，判断首尾是否相同
class Solution{
public:
    bool isPalindrome(int x) {
        if(x<0)
            return 0;
        string s = to_string(x);
        int i = 0;
        int j = s.length()-1;
        while(i<j)
        {
            if(s[i]!=s[j])
                return false;
            i++;
            j--;
        }
        return true;
    }
};

